### PR TITLE
add a version of `Branch.before` that can use a custom LCA implementation

### PR DESCRIPTION
**Pull request by aryairani**
Then it uses that new function `Branch.before'` in `SqliteCodebase.pushGitRootBranch` to use the sql lca implementation, speeding up the `before` operation by ~50% apparently.  I'm surprised it wasn't by more, but this is still nice.

### CODE REVIEW

**Comment by pchiusano** [comment]
@aryairani filled the rest of this in. Tests and transcripts pass. I haven't investigated how much faster it is though.

**Comment by aryairani** [comment]
It looks like it saves ~8 seconds on a push to an existing base repo; this transcript:
````
```ucm:hide
.> builtins.merge
.> builtins.mergeio
.> pull /root/base_v2:.trunk base
.> fork base newbase
.> merge builtin newbase
.> cd newbase
.newbase> push /root/unison-slim/repo
.newbase> delete.term Either.mapRight#jruvg9gh6q
.newbase> delete.term List.map#j1ejquc7so
.newbase> push /root/unison-slim/repo
```````

before PR (017b95826):
```
017b95826.log:GitRootBranch Branch.before in 16.170800121s
017b95826.log:GitRootBranch Branch.before in 16.631048373s
017b95826.log:GitRootBranch Branch.before in 16.67133445s
```
after PR (9e59b0ff1):
```
9e59b0ff1.log:GitRootBranch Branch.before in 8.077651517s
9e59b0ff1.log:GitRootBranch Branch.before in 7.927159894s
9e59b0ff1.log:GitRootBranch Branch.before in 7.966183706s
```

**Comment by aryairani** [comment]
@pchiusano I guess I should review it, because you implemented it; but because I had opened the PR, I can't review it, but you can.

**Review by pchiusano — APPROVED** [review]
Sweet! Go team. I wasn't sure if you wanted to do more review so I'll just approve and you can do the merge if everything looks good.

**Comment by pchiusano** [comment]
Oh nvm I see you already applied ready-to-merge.

**Pull request merged**
